singularity-eos/eos/eos_base.hpp

```diff
@@ -128,6 +128,13 @@ class EosBase {
   struct is_raw_pointer
       : std::is_same<std::remove_reference_t<std::remove_cv_t<T>>, R *> {};
 
+  // Generic evaluator
+  template <typename Functor_t>
+  PORTABLE_INLINE_FUNCTION void Evaluate(Functor_t &f) const {
```

Comment:
I changed this to accept non-const `operator()`s. This lets you create functors with side-effects in case you want to, e.g., store the value of a reduction.

I think const is less error prone, but this seems more flexible to me. Willing to walk it back if people object.

Reply:
Yeah I agree -- `const` is probably safer, but the ability to do reductions may come in handy and I think justifies dropping the `const` requirement.